### PR TITLE
Structure_oM: Add modifier fragment to Section and Surface Properties

### LINKED ISSUE #578: Structure_oM: Add modifier fragment class for properties
#### Description:

Modifiers for sections and surface properties are currently handled by custom data. This should be formalised into two Modifier classes:

Fragment used to adjust various properties of the section going into analysis packages

#### SectionModifiers:

- Area
- Iy
- Iz
- J
- Asy
- Asz

#### SurfaceModifiers:

- Nxx
- Nyy
- Nxy
- Mxx
- Myy
- Mxy
- Vxz
- Vyz

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### Issues addressed by this PR

Closes #578

Adding fragment to store modifiers information for sections and surface properties.

### CODE REVIEW

File: Structure_oM/Fragments/SurfacePropertyModifier.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        [Ratio]
+        [Description("Modifier of the bending stiffness about the local y-axis. Value of the SurfaceProperty is multiplied by this value, hence 1 means no scaling applied.")]
+        public virtual double MYY { get; set; } = 1;
+
+        [Ratio]
+        [Description("Modifier of the in-plane twist stiffness. Value of the SurfaceProperty is multiplied by this value, hence 1 means no scaling applied.")]
+        public virtual double MXY { get; set; } = 1;
+
+        [Ratio]
+        [Description("Modifier of the out of plane shear stiffness along the local x-axis. Value of the SurfaceProperty is multiplied by this value, hence 1 means no scaling applied.")]
```
Comment (peterjamesnugent) [inline Structure_oM/Fragments/SurfacePropertyModifier.cs:58]: Could we make the description clearer so it lines up with the subscript? Some conventions are the first subscript relates to the face it is acting on, and the second subscript is the direction it is acting like this:
image

File: Structure_oM/Fragments/SurfacePropertyModifier.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        public virtual double VXZ { get; set; } = 1;
+
+        [Ratio]
+        [Description("Modifier of the out of plane shear stiffness along the local y-axis. Value of the SurfaceProperty is multiplied by this value, hence 1 means no scaling applied.")]
```
Comment (peterjamesnugent) [inline Structure_oM/Fragments/SurfacePropertyModifier.cs:62]: As above

File: Structure_oM/Fragments/SurfacePropertyModifier.cs (hunk 1 above)
Comment (IsakNaslundBh) [reply Structure_oM/Fragments/SurfacePropertyModifier.cs:58]: agree. Have any suggestion to make it clearer?

Comment (peterjamesnugent) [reply Structure_oM/Fragments/SurfacePropertyModifier.cs]: Modifier of the out of plane shear stiffness along the local x-axis on the z-face. Whereby, the Z face is defined by it's normal, the local z-axis. Value of the SurfaceProperty is multiplied by this value, hence 1 means no scaling applied.

Seems very wordy...

**Comment by IsakNaslundBh** [comment]
Updating from master to try to please the CI

**Comment by IsakNaslundBh** [comment]
With SAP2000 PR now approved, this is now ready to be merged. @al-fisher

**Pull request merged**